10.7 Release Notes (#3388)
owncloud/docs: modules/admin_manual/pages/release_notes.adoc

* xref:changes-in-8-0[Changes in 8.0]
* xref:changes-in-7-0[Changes in 7.0]

== Changes in 10.7.0

Dear ownCloud administrator, please find below the changes and known issues in ownCloud Server 10.7 that need your attention. You can also read {owncloud-changelog-url}[the full ownCloud Server changelog] for further details on what has changed.

=== Migrations

> 

Reviewer: ```suggestion
```

=== Usability Improvements for the Classic Web Interface

> ==== Tagging

Reviewer: ```suggestion

==== Tagging
```

Based on user feedback, the user interface for file/folder tagging has been improved. Research has shown that some users misinterpret the tagging input field in the right sidebar as the location to share files/folders. To prevent this, the functionality to manage tags has been separated from the location where tags are displayed. From version 10.7 onwards, tags will be displayed as a read-only list at the top of the right sidebar. Tag management (assign/unassign, etc.) can now be conducted in a dedicated tab "Tags" below.

> ==== Sharing via E-mail Address

Reviewer 2: ```suggestion
==== Sharing via Email Address
```

Based on user feedback, some further usability improvements have been made in the user sharing dialog. In most cases users actually want to share with an internal or a guest user. Therefore, when sharing data by entering an email address, the suggestion to create a federated share will now only be shown when the internal user lookup has finished and did not return any results. As federated user IDs have the same syntax like email addresses (e.g., _userid@cloud.tld_) this measure has been taken to guide users better by only suggesting federated users when no other known users have been found.

==== Improvements for Opening Files with Multiple Applications

> Server 10.7 brings improvements for users when there are files that have multiple editor or viewer applications associated. For example, this could be a PDF file that can be opened with the integrated PDF viewer or with a third party application like Collabora Online or OnlyOffice. Previously the classic ownCloud Web interface would just make the decision for the user on click or require the user to use the respective entry in the file's dropdown menu. Starting with version 10.7, the user will be prompted on click to decide which application they want to use. If there is anyway only a single application available that is capable of opening the file, the behavior stays as before.

Reviewer 3: ```suggestion
Server 10.7 brings improvements for users when there are files that have multiple editor or viewer applications associated. For example, this could be a PDF file that can be opened with the integrated PDF viewer or with a third party application like Collabora Online or OnlyOffice. Previously the classic ownCloud Web interface would just make the decision for the user on click or require the user to use the respective entry in the file's dropdown menu. Starting with version 10.7, the user will be prompted on click to decide which application they want to use. If there is only a single application available that is capable of opening the file, the behavior stays as before.
```

=== Improvements for Storage Encryption

> Version 10.7 brings improvements for storage encryption in order to reduce storage usage. By changing from `base64` to `binary` encoding for encrypted files, an overhead of about 35% in storage usage can be saved. 

Reviewer: ```suggestion
Version 10.7 brings improvements for storage encryption in order to reduce storage usage. By changing from `base64` to `binary` encoding for encrypted files, a reduction of about 35% in storage usage can be achieved. 
```

> For existing installations that use storage encryption, this process is seamless. Files that have been stored before upgrading to 10.7 will stay with the previous encoding until they are modified which will store them with the new encoding.

Reviewer: ```suggestion
For existing installations that use storage encryption, this process is seamless. Files that have been stored before upgrading to 10.7 will stay with the previous encoding until they are rewritten which will store them with the new encoding.
```

=== Deprecation Note for User-key Storage Encryption

Storage encryption in ownCloud offers two options, master-key and user-key encryption. While master-key encryption is based on a general encryption key that is used to decrypt all user data, user-key encryption relies in essence on user passwords to decrypt individual user data. Both follow the goal to prevent malicious administrators from being able to read user data.

> Due to the nature of user-key storage encryption, this encryption mode comes with a list of https://doc.owncloud.com/server/latest/admin_manual/configuration/files/encryption/enabling-user-key-encryption.html#limitations[limitations] and can cause challenges for administrators, e.g., when users forget their password. 

Reviewer: ```suggestion
Due to the nature of user-key storage encryption, this encryption mode comes with a list of xref:configuration/files/encryption/enabling-user-key-encryption.adoc#limitations[limitations] and can cause challenges for administrators, e.g., when users forget their password. 
```

For these reasons, user-key storage encryption is now marked as deprecated and will not be maintained anymore for future versions of ownCloud Server. Server 10.7 still supports user-key encryption but the feature will be removed in later versions. If you are operating an ownCloud installation with user-key storage encryption enabled, please get in contact with support@owncloud.com to plan a migration to master-key storage encryption.

> TIP: Master-key storage encryption is still supported and has received improvements with Server 10.7 (see above). This encryption mode can be used with dedicated https://doc.owncloud.com/server/latest/admin_manual/configuration/server/security/hsmdaemon/[HSM products] for additional security.

Reviewer: ```suggestion
TIP: Master-key storage encryption is still supported and has received improvements with Server 10.7 (see above). This encryption mode can be used with dedicated xref:configuration/server/security/hsmdaemon.adoc[HSM products] for additional security.
```

=== Other Notable Changes

> * Redis can now be connected with TLS support for improved security. See the https://doc.owncloud.com/server/admin_manual/configuration/server/config_sample_php_parameters.html#define-redis-connection-details[documentation] for more information. https://github.com/owncloud/core/pull/38386[#38386]

Reviewer: ```suggestion
* Redis can now be connected with TLS support for improved security. See the xref:configuration/server/config_sample_php_parameters.adoc#define-redis-connection-details[documentation] for more information. https://github.com/owncloud/core/pull/38386[#38386]
```

> * For strong security, ownCloud Server uses strict same-site cookie handling. In certain scenarios (e.g., integrations) this behavior is not desired. To be able to flexibly adapt the intended behavior, https://doc.owncloud.com/server/latest/admin_manual/configuration/server/config_sample_php_parameters.html#define-how-to-relax-same-site-cookie-settings[the same-site cookie handling can now be configured]. https://github.com/owncloud/core/pull/38458[#38458]

Reviewer: ```suggestion
* For strong security, ownCloud Server uses strict same-site cookie handling. In certain scenarios (e.g., integrations) this behavior is not desired. To be able to flexibly adapt the intended behavior, the xref:configuration/server/config_sample_php_parameters.adoc#define-how-to-relax-same-site-cookie-settings[same-site cookie handling] can now be configured. https://github.com/owncloud/core/pull/38458[#38458]
```

* Loading the "Shared with you" list when shares originate from files on unavailable storages (e.g., Windows Network Drives) has been fixed. https://github.com/owncloud/core/pull/38190[#38190]
* Performance improvements for the "Shared with you" view list have been made. https://github.com/owncloud/core/pull/38385[#38385]
* Existing guest users are now correctly labeled as 'Guest' in the sharing sidebar tab (before they were labeled as regular 'User'). https://github.com/owncloud/core/pull/38440[#38440]
* Issues with multiple files with the same name e.g., in the "Shared by link" view, have been fixed https://github.com/owncloud/core/pull/38415[#38415]
* Error messages when uploading files are more specific again (e.g., when a virus has been found or when an upload was blocked due to a File Firewall rule) https://github.com/owncloud/core/pull/38416[#38416]

> * The https://doc.owncloud.com/server/admin_manual/configuration/server/occ_command.html#mimetype-update-commands[occ command] `maintenance:mimetype:update-db --repair-filecache` has been fixed. It can be used to bring changed mimetype associations for files into operation. https://github.com/owncloud/core/issues/38425[#38425]

Reviewer: ```suggestion
* The xref:configuration/server/occ_command.adoc#mimetype-update-commands[occ command] `maintenance:mimetype:update-db --repair-filecache` has been fixed. It can be used to bring changed mimetype associations for files into operation. https://github.com/owncloud/core/issues/38425[#38425]
```

* Subfolders from Google Drive can now be mounted. https://github.com/owncloud/core/pull/38161[#38161]
* The right scrollbar in the web interface is more visible now. https://github.com/owncloud/core/pull/38183[#38183]
* The user experience when adding external storages has been improved by clearly indicating success or errors.  https://github.com/owncloud/core/pull/38288[#38288]
* There are user groups which cannot be edited manually in user management (e.g., the `guests_app` group). These groups will not allow adding users anymore to prevent confusion. https://github.com/owncloud/core/pull/38298[#38298]
* Some unnecessary errors in the logs have been removed for cleanliness. https://github.com/owncloud/core/pull/38390[#38390]

=== Solved known issues

> Both https://doc.owncloud.com/server/admin_manual/release_notes.html#known-issues[known issues from Server 10.6] have been fixed.

Reviewer: ```suggestion
Both xref:release_notes.adoc#known-issues[known issues from Server 10.6] have been fixed.
```
